eventdata/parameter_sources/elasticlogs_kibana_source.py

```diff
@@ -190,9 +192,13 @@ def params(self):
             "dashboard": self._dashboard,
             "window_length": self._window_length,
             "ignore_throttled": self._ignore_throttled,
-            "pre_filter_shard_size": self._pre_filter_shard_size,
             "debug": self._debug
         }
+        request_params = {
```

Comment:
Alternatively this dict could be embedded in the `meta_data` dict as this is what we document as being passed for metrics in the `kibana_runner.py` signature (line 51). So it could look like:

```
        request_params = {
            "pre_filter_shard_size": self._pre_filter_shard_size
        }

        if self._max_concurrent_shard_requests > 0:
            request_params["max_concurrent_shard_requests"] = self._max_concurrent_shard_requests

        meta_data = {
            "interval": interval,
            "index_pattern": index_pattern,
            "query_string": query_string,
            "dashboard": self._dashboard,
            "window_length": self._window_length,
            "ignore_throttled": self._ignore_throttled,
            "debug": self._debug,
            "request_params": request_params
        }
```

and then we can delete `response["request_params"] = request_params` in `elasticlogs_kibana_sources.py#67` and change line 72 there to be:
`    result = await es.msearch(body=request, params=meta_data["request_params"])`

This could make the unit tests simpler i.e. only adding a field to test in `meta_data` but one could argue the current implementation is purer (request_params are request_params and not part of a meta_data object).

The actual metrics pushed would look the same in either case i.e.:

![image](https://user-images.githubusercontent.com/1754575/113166375-633bd480-924b-11eb-9ffd-9fa56e4bcb10.png)

Reply:
I prefer to separate the two: `request_params` is just like `body`, something we pass verbatim to the search. meta_data should be stuff that's not captured by either of them.